src/rust/engine/process_executor/src/lib.rs

```diff
@@ -0,0 +1,129 @@
+use std::io::Error;
+use std::process::Command;
+
+///
+/// A process to be executed.
+///
+/// # Example
+///
+/// ```
+/// use process_executor::ExecuteProcessRequest;
+/// let request = ExecuteProcessRequest {
+///     argv: vec!["/bin/bash".to_string(), "-c".to_string(), "/bin/echo Hello ${WHO}".to_string()],
+///     env: vec![("WHO".to_string(), "world".to_string())],
+/// };
+/// ```
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct ExecuteProcessRequest {
+    /// The arguments to execute. The first argument should be an absolute or relative path to the
+    /// binary to execute.
+    ///
+    /// No shell expansion or PATH lookup will be performed.
+    pub argv: Vec<String>,
+    /// The environment variables to set, in (key, value) pairs. Each key should appear at most one
+    /// time in the list.
+    ///
+    /// No other environment variables will be set for the execution (except possibly for an empty
+    /// PATH variable).
+    ///
+    /// Ideally this would be a HashMap, but that isn't hashable, and we need to implement Hash to
+    /// be in the node graph.
+    pub env: Vec<(String, String)>,
+}
+
+
+///
+/// The result of running a process.
+///
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ExecuteProcessResult {
+    pub stdout: Vec<u8>,
+    pub stderr: Vec<u8>,
+    pub exit_code: i32,
```

Comment:
`u8`?

Reply:
Windows, it turns out, has much more expressive exit codes... I don't see a good reason to throw away that information at the moment, especially as all calling code should just be calling "exit_code == 0", and if this ends up on the Python side of a boundary it doesn't make a difference anyway.